app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt

```diff
@@ -42,24 +42,20 @@ class MainFragment : Fragment() {
     private val SAVE_IMAGE_REQUEST_CODE: Int = 1999
     private val CAMERA_REQUEST_CODE: Int = 1998
     private val AUTH_REQUEST_CODE = 2002
-    val CAMERA_PERMISSION_REQUEST_CODE = 1997
-
-    private lateinit var currentPhotoPath: String
-    lateinit var navController: NavController
-
-    protected var puzzleURI : Uri? = null
+    private val CAMERA_PERMISSION_REQUEST_CODE = 1997
     private var puzzles : ArrayList<Picture> = ArrayList<Picture>()
-
     private var _binding: MainFragmentBinding? = null
     private val binding get() = _binding!!
-
     private var user : FirebaseUser? = null
+    private lateinit var currentPhotoPath: String
+    protected var puzzleURI : Uri? = null
+    private lateinit var viewModel: MainViewModel
+    lateinit var navController: NavController
```

Comment:
Ordering the modifiers in the correct order.
Source: https://kotlinlang.org/docs/coding-conventions.html#modifiers-order